colossalai/pipeline/schedule/_utils.py

```diff
@@ -118,15 +149,18 @@ def merge_batch(data: List[Any]) -> Any:
     flattened_data = []
     tree_spec = None
     for d in data:
-        elems, tree_spec = tree_flatten(d)
+        # elems should be an instance of OrderedDict
+        elems, tree_spec = tree_flatten_hf(d)
```

Comment:
Do you need to update `_detach()` with `tree_flatten_hf()`?

Reply:
this is done with `tree_map_hf` ,in line 10 ,and the corresponding changes also in one_f one_b